03-rnaseq/dimension-reduction_rnaseq_02_umap.Rmd

```diff
@@ -0,0 +1,414 @@
+  tibble::rownames_to_column("refinebio_accession_code") %>% # Let's get the rownames into a column so we can merge the annotation data
+  dplyr::inner_join(metadata, by = "refinebio_accession_code") # Let's merge the annotation data using the `refinebio_accession_code` column
+```
+
+## Create UMAP plot
+
+Now we can use the `ggplot` function to plot our normalized UMAP scores.
+
+```{r}
+# Plot using `ggplot()` function
+ggplot(umap_plot_df, aes(x = X1, y = X2)) +
+  geom_point() # This tells R that we want a scatterplot
+```
+
+Let's try adding a variable to our plot for annotation.
+
+In this code chunk, the variable `refinebio_treatment` is given to the `ggplot()` function as part of the goal of the experiment is to analyze the sample transcriptional responses to androgen deprivation therapy (ADT).
+
+```{r}
+# Plot using `ggplot()` function
+ggplot(
+  umap_plot_df,
+  aes(
+    x = X1,
+    y = X2,
+    color = refinebio_treatment # Here we are giving `refinebio_treatment` to the color argument
+  )
+) +
+  geom_point() # This tells R that we want a scatterplot
+```
+
+In the next code chunk, we are going to add another variable to our plot for annotation.
+
+We'll plot using both `refinebio_treatment` and `refinebio_disease` variables for labels since they are central to the androgen deprivation therapy (ADT) based hypothesis in the [original paper](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6210624/) [@ADT-Prostate-Cancer].
+
+```{r}
+# Plot using `ggplot()` function
+final_annotated_umap_plot <- ggplot(
+  umap_plot_df,
+  aes(
+    x = X1,
+    y = X2,
+    color = refinebio_treatment,
+    # Here we are giving `refinebio_treatment` to the color argument
+    shape = refinebio_disease # Here we are giving `refinebio_disease` to the shape argument
+  )
+) +
+  geom_point() # This tells R that we want a scatterplot
+
+# Display the plot that we saved above
+final_annotated_umap_plot
+```
+
+The plot above is more informative than the first plot we created but is still not as informative as one may home as there are no clear clusters represented.
+Although it does appear that majority of the pre-ADT and post-ADT appear to cluster together, there are still questions remaining as we look at outliers.
+
+### Interpretation of UMAP plot and results
+
+1. Note that the coordinates of UMAP output for any given cell can change dramatically depending on parameters, and even run to run with the same parameters. 
+This means that you should not rely too heavily on the exact values of UMAP's output. 
+
+  - One particular limitation of UMAP is that while observed clusters have some meaning, the distance *between* clusters usually does not (nor does cluster density). The fact that two clusters are near each other should NOT be interpreted to mean that they are more related to each other than to more distant clusters. (There is some disagreement about whether UMAP distances have more meaning, but it is also probably safer to assume they don't.)
```

Comment:
I do have a question about whether we need to cite ourselves from training-modules? Not sure how this works.

Reply:
I attempted to do this, but not sure I _fully_ followed the citation guidelines? Wasn't completely sure who the author should be (`CCDL 2020` as in the notebook or those who I can see contributed in Git history).

Reply:
This is a tricky instance. I think what you have now is fine.